becquerel/core/fitting.py

```diff
@@ -0,0 +1,537 @@
+from __future__ import print_function
+import numpy as np
+import pandas as pd
+import scipy.special
+from lmfit.model import Model
+import matplotlib.pyplot as plt
+from matplotlib.gridspec import GridSpec
+from matplotlib.font_manager import FontProperties
+
+
+FWHM_SIG_RATIO = 2.35482
+
+
+class FittingError(Exception):
+    """Exception raised by Fitters."""
+    pass
+
+
+# -----------------------------------------------------------------------------
+# Base functions
+# -----------------------------------------------------------------------------
+
+
+def constant(x, c):
+    return np.ones_like(x) * c
+
+
+def line(x, m, b):
+    return m * x + b
+
+
+def gauss(x, amp, mu, sigma):
+    amp = np.abs(amp)
+    mu = np.abs(mu)
```

Comment:
Is there any reason to restrict `mu` to non-negative values? It's probably usually true for spectral peaks, but this feels more restrictive than robust.

Reply:
Good point. I suppose not necessarily. This should be a higher level configuration specifically for peak fitting in the constraints.